src/rust/engine/process_execution/src/nailgun/mod.rs

```diff
@@ -213,7 +214,7 @@ impl super::CommandRunner for CommandRunner {
 }
 
 #[async_trait]
-impl CapturedWorkdir for CommandRunner {
+impl<T: LocalCommandRunner> CapturedWorkdir for CommandRunner<T> {
```

Comment:
Interesting... I'm not sure what it would look like to interact via nailgun with a process running inside of docker. But it's not clear what `LocalCommandRunner` is actually used for, so.

Reply:
It exists solely as temporary scaffolding so that the nailgun runner can continue to access the struct's it was accessing directly in the `local::ComandRunner` without me having to reason at this moment about what the ultimate way to modify the nailgun runner is.

Your advice on what to do with the nailgun runner or better modifying the interaction between the nailgun runner and the local runners to not require access to those struct's will be useful.

Reply:
For now, would it be better to disable nailgun when Docker is enabled? Or add some minimal form of environment support that could switch between runners?

Reply:
> Or add some minimal form of environment support that could switch between runners?

Right, possibly. The global switch doesn't make sense.

My expectation is that in production usage of this feature, the `Context::command_runners` field: https://github.com/pantsbuild/pants/blob/da575b0e28dad35a99e4880892dcde3833b7bbca/src/rust/engine/src/context.rs#L64-L67 ...might become a `HashMap` from "environment matcher" to `CommandRunner`s.

I don't know how to minimally hack something like that in here, except perhaps to mark a `Process` struct with something that indicates that docker should be used. Sometime this week I'll probably start pushing on having `Process` consume the current environment, but for now, you could maybe use an env-var, and then switch between command_runners here: ...? https://github.com/pantsbuild/pants/blob/da575b0e28dad35a99e4880892dcde3833b7bbca/src/rust/engine/src/nodes.rs#L392-L402

Reply:
Latest iteration removes the integration entirely and just relies on tests. So we can punt on the integration issue to a future PR.